<launch>
    <env name="ROSCONSOLE_CONFIG_FILE" value="$(find exploration)/cfg/custom_rosconsole.config"/>

    <arg name="robot_name" default="robot1"/>

    <arg name="map" default="/$(arg robot_name)/map"/>
    <!-- <arg name="map" default="/server/grid_map_merge/merge_map"/> -->
    <arg name="pose" default="/$(arg robot_name)/robot_pose"/>

    <arg name="velocity" default="/$(arg robot_name)/mobile_base/commands/velocity"/>
    <arg name="scan" default="/$(arg robot_name)/scan"/>
    <arg name="bumper_raw" default="/$(arg robot_name)/mobile_base/events/bumper"/>
    <arg name="map_frame_id" default="$(arg robot_name)/map"/>

    <arg name="rosbag" default="false"/>
    <arg name="debug" default="false"/>

    <arg name="fill" default="true"/>

    <arg name="ns" default="exploration"/>

    <group ns="$(arg robot_name)/$(arg ns)">
        <node pkg="exploration" type="frontier_based_exploration" name="frontier_based_exploration" output="screen">
            <remap unless="$(arg fill)" from="map" to="$(arg map)"/>
            <remap if="$(arg fill)" from="map" to="fill_map"/>
            <remap from="pose" to="$(arg pose)"/>
            <remap unless="$(arg debug)" from="velocity" to="$(arg velocity)"/>
            <param name="map_frame_id" value="$(arg map_frame_id)"/>
            <param name="movebase_name" value="/$(arg robot_name)/move_base"/>
            <param name="debug" value="$(arg debug)"/>
            <param name="direction_weight" value="0.0"/>
            <param name="min_cluster_size" value="35"/>
            <param name="previous_goal_effect" value="false"/>
            <!-- <param name="merge_map_frame_id" value="/server/merge_map"/> -->
            <!-- <param name="use_merge_map" value="true"/> -->
        </node>

        <node pkg="exploration" type="visualization" name="visualization">
            <param name="map_frame_id" value="$(arg robot_name)/map"/>
        </node>

        <node unless="$(arg debug)" pkg="exploration" type="log" name="log">
            <remap from="pose" to="$(arg pose)"/>
        </node>

        <node pkg="exploration" type="map_fill" name="map_fill">
            <remap from="map" to="$(arg map)"/>
        </node>

        <node pkg="exploration" type="exploration_manager" name="exploration_manager">
            <remap unless="$(arg fill)" from="map" to="$(arg map)"/>
            <remap if="$(arg fill)" from="map" to="fill_map"/>
        </node>

	    <include if="$(arg rosbag)" file="$(find exploration)/launch/includes/rosbag_record.launch.xml">
            <!-- <arg name="dir" value="/home/rosss1/rosbag_debug"/> -->
            <arg name="topics" value="  $(arg pose) 
                                        /$(arg robot_name)/$(arg ns)/pose_log 
                                        $(arg bumper_raw) 
                                        $(arg scan) 
                                        $(arg velocity) 
                                        /$(arg robot_name)/$(arg ns)/visualization 
                                        /tf 
                                        /tf_static 
                                        /$(arg robot_name)/map 
                                        /$(arg robot_name)/move_base/global_costmap/costmap 
                                        /$(arg robot_name)/NavfnROS/plan"/>
        </include>

    </group>
</launch>
